core/src/main/scala/org/apache/spark/deploy/history/HistoryServer.scala

```diff
@@ -143,6 +143,12 @@ class HistoryServer(
     appCache.stop()
   }
 
+  // For testing - override stop timeout used by jetty
+  private[history] def setStopTimeout(timeout: Long): Unit = {
```

Comment:
I think that adding a function just to speed up tests by 30sec is unnecessary additional code in this case